jenkins/artifact-signer/Jenkinsfile

```diff
@@ -0,0 +1,72 @@
+lib = library(identifier: 'jenkins@20211123', retriever: legacySCM(scm))
+
+pipeline {
+    agent {
+        docker {
+            label 'Jenkins-Agent-al2-x64-c54xlarge-Docker-Host'
+            image 'opensearchstaging/ci-runner:ci-runner-centos7-v1'
+            alwaysPull true
+        }
+    }
+    parameters{
+        string(
+                name: 'ARTIFACT_PATH',
+                trim: true,
+                description: 'Path to the artifact or the artifact directory on s3'
+        )
+        choice(
+                choices: ['linux'],
+                name: 'DISTRIBUTION_PLATFORM',
+                description: 'What platform is this distribution build for?'
+        )
+        choice(
+                choices: ['.sig'],
+                name: 'SIGNATURE_TYPE',
+                description: 'What is signature file type?'
+        )
+    }
+    stages {
+        stage('sign') {
+            steps {
+                script {
+                    if (ARTIFACT_PATH == '') {
+                        currentBuild.result = 'ABORTED'
+                        error('ARTIFACT_PATH is not set')
+                    }
+                    ARTIFACT_PATH = ARTIFACT_PATH.replaceAll('/$', "")
+                    set_up()
+                    signArtifacts(
+                            artifactPath: "$WORKSPACE/artifacts/${ARTIFACT_PATH}",
+                            signatureType: SIGNATURE_TYPE,
+                            distributionPlatform: DISTRIBUTION_PLATFORM
+                    )
+                    upload_artifacts()
+                }
+            }
+            post() {
+                always {
+                    script {
+                        postCleanup()
+                    }
+                }
+            }
+        }
+    }
+}
+
+void upload_artifacts(){
+    // Upload signature files to S3
+    withAWS(role: "${ARTIFACT_UPLOAD_ROLE_NAME}", roleAccount: "${AWS_ACCOUNT_PUBLIC}", duration: 900, roleSessionName:
+            'jenkins-session') {
+        // upload signature files to S3
+        s3Upload(file: "$WORKSPACE/artifacts/${ARTIFACT_PATH}", bucket: "${ARTIFACT_BUCKET_NAME}", path: "${ARTIFACT_PATH}/signed")
+    }
+}
+
+void set_up() {
```

Comment:
This downloads unsigned artifacts, so should be called `downloadArtifacts()` to match `uploadSignedArtifacts()`

Reply:
renamed to `setUpSignArtifacts`